### DISCUSSION TITLE
dunder methods and possible future

### DISCUSSION THREAD

**Discussion by jonas-eschle** [pydantic/pydantic#4573] [Pydantic V2]
Hi there, I came across the multiple usages of custom dunder methods in pydantic. These methods are Python "reserved" namespace and [new Python methods can be introduced anytime](https://docs.python.org/3/reference/lexical_analysis.html#reserved-classes-of-identifiers), i.e. they are listed under reserved namespace.

Therefore, using custom dunder methods seems like bad practice. Any plans to move to something safe, i.e. `_*_` for v2?

**Comment by samuelcolvin** [comment]
Humm, maybe.

I've used them quite a lot so far I'm v2, but maybe we should remove them. 

I think it's fine use dunder attributes starting "pydantic", e.g. `__pydantic_validator__`. The chance of them being adopted by cpython is virtually zero.

**Reply by jonas-eschle** [reply]
Yes, I agree, that seems to be the convention that many libraries go with and unambiguous
